internal/registry/registry_cache.go

```diff
@@ -115,6 +120,47 @@ func (r *Cache) LocateBuildpack(bp string) (Buildpack, error) {
 	return Buildpack{}, fmt.Errorf("no entries for buildpack: %s", bp)
 }
 
+// LocateBuildpack stored in registry
+func (r *Cache) LocateExtension(ex string) (Buildpack, error) {
```

Comment:
If we do the above suggestion we could make a private helper like `locateModule(kind string, ex string)` with different invocations for buildpack and extension (see for example builder.go `addModules`)